numpy/core/meson.build

```diff
@@ -709,9 +718,255 @@ foreach gen: test_modules_src
     dependencies: np_core_dep,
     install: true,
     subdir: 'numpy/core',
+    link_with: gen[2],
   )
 endforeach
 
+# Build multiarray dispatch-able sources
+# --------------------------------------
+multiarray_gen_headers = [
+  src_file.process('src/multiarray/arraytypes.h.src'),
+  src_file.process('src/common/npy_sort.h.src'),
+]
+multiarray_umath_dispatch_link = []
+foreach gen_mtargets : [
+  [
+    'argfunc.dispatch.h',
+    src_file.process('src/multiarray/argfunc.dispatch.c.src'),
+    [
+      AVX512_SKX, AVX2, XOP, SSE42, SSE2,
+      VSX2,
+      ASIMD, NEON,
+      VXE, VX
+    ]
+  ],
+]
+mtargets_res = mod_features.multi_target(
+  gen_mtargets[0], multiarray_gen_headers + gen_mtargets[1],
+  dispatch: gen_mtargets[2],
+  baseline: CPU_BASELINE,
+  prefix: 'NPY_',
+  dependencies: [py_dep, np_core_dep],
+  c_args: c_args_common + max_opt,
+  cpp_args: cpp_args_common + max_opt,
+  include_directories: [
+    'include',
+    'src/common',
+    'src/multiarray',
+    'src/npymath',
+    'src/umath'
+  ]
+)
+CPU_MTARGETS_LOG += {gen_mtargets[0]: mtargets_res[0]}
+multiarray_umath_dispatch_link += mtargets_res[1]
+endforeach
+
+# Build npysort dispatch-able sources
+# -----------------------------------
+foreach gen_mtargets : [
+  [
+    'simd_qsort.dispatch.h',
+    'src/npysort/simd_qsort.dispatch.cpp',
+    [AVX512_SKX]
+  ],
+  [
+    'simd_qsort_16bit.dispatch.h',
+    'src/npysort/simd_qsort_16bit.dispatch.cpp',
+    [AVX512_SPR]
```

Comment:
Should this be `[AVX512_ICL, AVX512_SPR]` ?

Reply:
Indeed, done.